src/main/java/io/reactivesw/category/domain/service/update/SetOrderHintService.java

```diff
@@ -4,23 +4,50 @@
 import io.reactivesw.category.domain.model.Category;
 import io.reactivesw.category.infrastructure.update.UpdateAction;
 import io.reactivesw.category.infrastructure.util.CategoryActionUtils;
+import io.reactivesw.category.infrastructure.util.CategoryUtils;
+import io.reactivesw.category.infrastructure.validator.CategoryOrderHintValidator;
 import io.reactivesw.model.Updater;
+
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Service;
 
+import java.math.BigDecimal;
+
 /**
- * Created by Davis on 16/12/29.
+ * Service to set orderHint when update category.
  */
 @Service(value = CategoryActionUtils.SET_ORDER_HINT)
 public class SetOrderHintService implements Updater<Category, UpdateAction> {
+
   /**
-   * set order hint.
+   * Set order hint.
    *
    * @param entity E
    * @param action UpdateAction
    */
   @Override
   public void handle(Category entity, UpdateAction action) {
     SetOrderHint setOrderHint = (SetOrderHint) action;
-    entity.setOrderHint(setOrderHint.getOrderHint());
+    CategoryOrderHintValidator.validateEmptyAndNumeric(setOrderHint);
+    //    Category is changed to the last one.
+    if (StringUtils.isBlank(setOrderHint.getNextOrderHint())) {
```

Comment:
In CategoryOrderHintValidator, when nextOrderHint is empty, nothing hanppen. It is ok.
By the way, I think isEmpty method is better than isBlank, So i decided to change back to isEmpty.